### DISCUSSION TITLE
How to read the github (or similar) url of current project

### DISCUSSION THREAD

**Discussion by msorens** [microsoft/vscode-discussions#425] [Extension Development QnA] [answered]
In my extension I need to know the GitHub URL (whether it be SSH or TLS) for the current project open in VSCode. I do not need to do an authentication handshake—I found the sample for that. I just need to know the URL. It seems silly to have to ask the user to type it in since it must clearly exist somewhere; just not finding any references to it.

**Comment by digitarald** [comment]
I would recommend checking out how the GitHub Pull Request extension is detecting the github remote in this and other files: https://github.com/microsoft/vscode-pull-request-github/blob/main/src/common/remote.ts

**Reply by olguzzar** [reply]
Hi @msorens, did checking out the above example help you with your issue?

**Reply by msorens** [reply]
It will likely help, but needs some digging, so it has become a bit lower priority for me.

**Comment by alefragnani** [comment, marked as answer]
Hi @msorens ,

Just out of curiosity, when you say _GitHub URL_, do you mean the _Git repo remote URL_, the Codespaces URL (in order to open the same project again), or other info?

I asked about Codespaces because I had to _decifer_ such info in order to be able to open _saved_ Codespaces projects on my Project Manager extension. 

Maybe that could help
